### PR TITLE
Ci rustfmt v1

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
None : QA

Describe changes:
- GitHub action to check that rustfmt-ed files are still rustfmt-ed
- rustfmt some more files which have a small diff

### CODE REVIEW

File: .github/workflows/formatting.yml [hunk 1]
```diff
@@ -126,6 +126,7 @@ jobs:
           fi
         shell: bash {0}
       - run: git clone https://github.com/OISF/libhtp -b 0.5.x
+      - run: ./qa/rustfmt.sh
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply .github/workflows/formatting.yml:129]: This is both formatting and rust, so might fit in both ?

**Comment by catenacyber** [comment]
https://redmine.openinfosecfoundation.org/issues/3836

File: qa/rustfmt.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+r=0
```
Comment (jasonish) [inline qa/rustfmt.sh:1]: Lets please make this a proper script.. Start with `#! /bin/sh`, little comment what it does.

File: qa/rustfmt.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+cat qa/rustfmt.txt | while read i; do
+    rustfmt $i;
```
Comment (jasonish) [inline qa/rustfmt.sh:3]: I'd just call `cargo fmt` then check each file.

File: qa/rustfmt.txt [hunk 4]
```diff
@@ -0,0 +1,57 @@
+rust/derive/src/applayerframetype.rs
```
Comment (jasonish) [inline qa/rustfmt.txt:1]: Maybe build this list within the context of the `rust/` directory, as in where its valid to run `cargo fmt`.

File: .github/workflows/formatting.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jasonish) [reply .github/workflows/formatting.yml:129]: Fair enough, keep it in formatting.

**Review by jasonish — CHANGES_REQUESTED** [review]
Good start.  While I'd rather just format `master` and `master-6.0.x` in one go at the same time, this is probably more realistic approach to get to whats expected of Rust projects today.

File: qa/rustfmt.sh (hunk 2 above)
Comment (catenacyber) [reply qa/rustfmt.sh:1]: ok

**Comment by victorjulien** [comment]
replaced by #9072

**Pull request closed without merging**